implementation/domain/logic/jbb-board-impl/src/main/java/org/jbb/board/impl/forum/logic/ForumServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+        Validate.notNull(category);
+
+        ForumCategoryEntity categoryEntity = categoryRepository.findOne(category.getId());
+
+        Integer lastPosition = getLastForumPosition(categoryEntity);
+
+        ForumEntity forumEntity = ForumEntity.builder()
+                .name(forum.getName())
+                .description(forum.getDescription())
+                .locked(forum.isLocked())
+                .position(lastPosition + 1)
+                .category(categoryEntity)
+                .build();
+
+        Set<ConstraintViolation<ForumEntity>> validationResult = validator.validate(forumEntity);
+
+        if (!validationResult.isEmpty()) {
+            throw new ForumException(validationResult);
+        }
+
+        forumEntity = forumRepository.save(forumEntity);
+
+        eventBus.post(new ForumCreatedEvent(forumEntity.getId()));
+
+        return forumEntity;
+    }
+
+    @Override
+    @Transactional
+    public Forum moveForumToPosition(Forum forum, Integer newPosition) {
+        Validate.notNull(forum);
+        Validate.notNull(newPosition);
+
+        ForumEntity movingForumEntity = forumRepository.findOne(forum.getId());
+        Integer oldPosition = movingForumEntity.getPosition();
+        ForumCategoryEntity categoryEntity = movingForumEntity.getCategory();
+
+        Validate.inclusiveBetween(1L, getLastForumPosition(categoryEntity), newPosition);
+        List<ForumEntity> allForums = forumRepository.findAllByCategoryOrderByPositionAsc(categoryEntity);
+        allForums.stream()
+                .filter(forumEntity -> forumEntity.getId().equals(movingForumEntity.getId()))
+                .forEach(movedForumEntity -> movedForumEntity.setPosition(-1));
+
+        allForums.stream()
+                .filter(forumEntity -> forumEntity.getPosition() > oldPosition)
+                .forEach(entity -> entity.setPosition(entity.getPosition() - 1));
+
+        allForums.stream()
+                .filter(forumEntity -> forumEntity.getPosition() >= newPosition)
+                .forEach(entity -> entity.setPosition(entity.getPosition() + 1));
+
+        allForums.stream()
+                .filter(forumEntity -> forumEntity.getId().equals(movingForumEntity.getId()))
+                .forEach(movedForumEntity -> movedForumEntity.setPosition(newPosition));
+
+        forumRepository.save(allForums);
+
+        return forumRepository.findOne(forum.getId());
+    }
+
+    @Override
+    @Transactional
+    public Forum moveForumToAnotherCategory(Long forumId, Long categoryId) {
+        Validate.notNull(forumId);
+        Validate.notNull(categoryId);
+
+        ForumEntity movingForumEntity = forumRepository.findOne(forumId);
+        ForumCategoryEntity newCategoryEntity = categoryRepository.findOne(categoryId);
+
+        ForumCategoryEntity currentCategoryEntity = movingForumEntity.getCategory();
+        currentCategoryEntity.getForumEntities().remove(movingForumEntity);
+        movingForumEntity.setCategory(null);
```

Comment:
This line can be removed. ForumCategoryEntity is a JPA "owning side", so removing forum from category is sufficient and "nullable" of category for forum entity is redundant.